cli/src/cmd/forge/script/cmd.rs

```diff
@@ -61,56 +63,77 @@ impl ScriptArgs {
             self.execute(&mut script_config, contract, sender, &predeploy_libraries).await?;
 
         if self.resume || (self.verify && !self.broadcast) {
-            let fork_url = self
-                .evm_opts
-                .fork_url
-                .clone()
-                .ok_or_else(|| eyre::eyre!("You must provide an RPC URL (see --fork-url)."))?;
-
-            let provider = Arc::new(try_get_http_provider(&fork_url)?);
-            let chain = provider.get_chainid().await?.as_u64();
-
-            verify.set_chain(&script_config.config, chain.into());
-
-            let broadcasted = self.broadcast || self.resume;
-            let mut deployment_sequence = match ScriptSequence::load(
-                &script_config.config,
-                &self.sig,
-                &target,
-                chain,
-                broadcasted,
-            ) {
-                Ok(seq) => seq,
-                // If the script was simulated, but there was no attempt to broadcast yet,
-                // try to read the script sequence from the `dry-run/` folder
-                Err(_) if broadcasted => {
-                    ScriptSequence::load(&script_config.config, &self.sig, &target, chain, false)?
-                }
-                Err(err) => eyre::bail!(err),
-            };
+            if !self.multi {
+                let fork_url =
+                    self.evm_opts.fork_url.clone().ok_or_else(|| {
+                        eyre::eyre!("You must provide an RPC URL (see --fork-url).")
+                    })?;
+
+                let provider = Arc::new(try_get_http_provider(&fork_url)?);
+                let chain = provider.get_chainid().await?.as_u64();
 
-            receipts::wait_for_pending(provider, &mut deployment_sequence).await?;
+                verify.set_chain(&script_config.config, chain.into());
 
-            if self.resume {
-                self.send_transactions(&mut deployment_sequence, &fork_url, result.script_wallets)
+                let broadcasted = self.broadcast || self.resume;
+                let mut deployment_sequence = match ScriptSequence::load(
+                    &script_config.config,
+                    &self.sig,
+                    &target,
+                    chain,
+                    broadcasted,
+                ) {
+                    Ok(seq) => seq,
+                    // If the script was simulated, but there was no attempt to broadcast yet,
+                    // try to read the script sequence from the `dry-run/` folder
+                    Err(_) if broadcasted => ScriptSequence::load(
+                        &script_config.config,
+                        &self.sig,
+                        &target,
+                        chain,
+                        false,
+                    )?,
+                    Err(err) => eyre::bail!(err),
+                };
+
+                receipts::wait_for_pending(provider, &mut deployment_sequence).await?;
+
```

Comment:
`ScriptArgs::run_script` is now >200 lines with large if else branches.

can we separate the `if self.resume || (self.verify && !self.broadcast)`condition into separate functions?